pandas/core/sample.py

```diff
@@ -150,6 +150,14 @@ def sample(
         else:
             raise ValueError("Invalid weights: weights sum to zero")
 
+    if weights is not None:
+        is_max_weight_dominating = size * weights.max() > 1
+        if is_max_weight_dominating and not replace:
+            raise ValueError(
+                "Invalid weights: If `replace`=False, "
+                "total unit probabilities have to be less than 1"
+            )
```

Comment:
- Avoid O(n) computation if it is not needed.
- Combine with the `if` block above; no need to check `if weights is not None` twice.
- Suggestion on a more clear error message.

```suggestion
        if not replace and size * weights.max() > 1:
            raise ValueError(
                "Weighted sampling cannot be achieved with replace=False. Either "
                "set replace=True or use smaller weights. See the docstring of sample "
                "for details."
            )
```